### PR TITLE
[iOS] Update save area on orientation

### LINKED ISSUE #1105: [Bug] Save area does not update on orientation and status bar changes
### Steps to Reproduce

1. Open sample app with safe area effect demo
2. Rotate phone while effect is enabled

### Expected Behavior

Simulator Screen Shot - iPhone 12 - 2021-03-22 at 13 53 39
Simulator Screen Shot - iPhone 12 - 2021-03-22 at 13 54 18

### Actual Behavior

Simulator Screen Shot - iPhone 12 - 2021-03-22 at 13 54 35
Simulator Screen Shot - iPhone 12 - 2021-03-22 at 13 53 55

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Workaround

Disable -> Enable effect after rotation.

### PR DESCRIPTION
**Pull request by geskill**
### Description of Change ###

Save area will be updated upon simulator/device orientation and status bar size changes (e.g. incoming call prior to iOS 14).

### Bugs Fixed ###

- Fixes #1105

### API Changes ###

None.

### Behavioral Changes ###

None.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by pictos** [comment]
@geskill I know that this a draft PR, but could you open an issue and attach a small repro for this problem? Thanks in advance

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SafeArea/SafeAreaEffectRouter.ios.cs [hunk 1]
```diff
@@ -25,23 +28,41 @@ protected override void OnAttached()
...
+		{
+			if (!IsEligibleToConsumeEffect)
+				return;
+
+			if (didChangeStatusBarOrientationNotificationObserver != null)
+				NSNotificationCenter.DefaultCenter.RemoveObserver(didChangeStatusBarOrientationNotificationObserver);
+			if (didChangeStatusBarFrameNotificationObserver != null)
+				NSNotificationCenter.DefaultCenter.RemoveObserver(didChangeStatusBarFrameNotificationObserver);
+
+			Element.Margin = initialMargin;
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SafeArea/SafeAreaEffectRouter.ios.cs:51]: should we dispose observers?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SafeArea/SafeAreaEffectRouter.ios.cs [hunk 2]
```diff
@@ -25,23 +28,41 @@ protected override void OnAttached()
 			if (!IsEligibleToConsumeEffect)
 				return;
 
+			didChangeStatusBarOrientationNotificationObserver = NSNotificationCenter.DefaultCenter.AddObserver(
+				UIApplication.DidChangeStatusBarOrientationNotification, (NSNotification n) => { UpdateInsets(); });
```
Comment (maxkoshevoi) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SafeArea/SafeAreaEffectRouter.ios.cs:32]: nit: 
```suggestion
				UIApplication.DidChangeStatusBarOrientationNotification, _ => UpdateInsets());
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SafeArea/SafeAreaEffectRouter.ios.cs [hunk 3]
```diff
@@ -25,23 +28,41 @@ protected override void OnAttached()
...
+
+			didChangeStatusBarFrameNotificationObserver = NSNotificationCenter.DefaultCenter.AddObserver(
+				UIApplication.DidChangeStatusBarFrameNotification, (NSNotification n) => { UpdateInsets(); });
```
Comment (maxkoshevoi) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SafeArea/SafeAreaEffectRouter.ios.cs:35]:
```suggestion
				UIApplication.DidChangeStatusBarFrameNotification, _ => UpdateInsets());
```

**Comment by geskill** [comment]
- Cut update on status bar changes because this is no longer changing in current iOS version.
- Use `UIDevice.OrientationDidChangeNotification` instead of deprecated `UIApplication.DidChangeStatusBarOrientationNotification`.

**Review by AndreiMisiukevich — APPROVED** [review]

**Comment by maxkoshevoi** [comment]
@AndreiMisiukevich Shouldn't this be merged in `main` as well, since it's a bugfix that doesn't change public API?

**Pull request merged**